packages/lib-react-components/src/FavoritesIconButton/FavoritesIconButtonContainer.js

```diff
@@ -0,0 +1,127 @@
+import { shape, string } from 'prop-types'
+import useSWRMutation from 'swr/mutation'
+
+import {
+  usePanoptesAuthToken,
+  useUserCollections
+} from '../hooks'
+import {
+  addSubjectsToCollection,
+  removeSubjectsFromCollection
+} from '../helpers/collections'
+import FavoritesIconButton from './FavoritesIconButton'
+
+function FavoritesIconButtonContainer({
+  project,
+  subject,
+  user,
+  ...props
+}) {
+  const token = usePanoptesAuthToken()
+
+  const query = {
+    favorite: true,
+    project_ids: [project?.id],
+    owner: user?.login
+  }
+
+  const { trigger: addToFavorites } = useSWRMutation({ query, token }, addSubjectsToCollection)
+  const { trigger: removeFromFavorites } = useSWRMutation({ query, token }, removeSubjectsFromCollection)
+
+  const {
+    data: favorites,
+    error,
+    isLoading
+  } = useUserCollections({
+    query
+  })
+
+  const isFavorite = favorites?.[0]?.links?.subjects?.includes(subject.id) ?? false
```

Comment:
Right now, the code is only checking if a subject is a favorite per the favorites collection, however, a subject from `/recents` or per the subject queue (in the classifier) includes a `favorite` boolean property. 

So before this FavoritesIconButton is used with `/recents` or in the classifier further refactoring could prevent the in those cases unnecessary request for the favorites collection. However, that would then complicate adding the subject to a favorites collection, because we'd have to first check if a favorites collection exists, then add the subject to it or create a new favorites collection with it. Nothing major, already done in [Collections](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/app-project/stores/User/Collections/Collections.js), but noting.